plan-for-extension-exercise/src/planForExtensionExercise/Shape.java

```diff
@@ -0,0 +1,9 @@
+package planForExtensionExercise;
+
+public abstract class Shape {
```

Comment:
I used an abstract class largely because the relationship between Shape and
its subclasses is a very strong, hierarchical one so it seemed to make
sense from an interpretability standpoint. Also it is my understanding that
abstract classes are faster than interfaces and may be preferable in cases
where there is no obvious advantage to using an interface.



On Thu, Feb 5, 2015 at 11:11 AM, Spencer Robinson <notifications@github.com>
wrote:

> In plan-for-extension-exercise/src/planForExtensionExercise/Shape.java
> <https://github.com/siostechcorp/exercises/pull/2#discussion_r24173904>:
>
> > @@ -0,0 +1,9 @@
> > +package planForExtensionExercise;
> > +
> > +public abstract class Shape {
>
> Because this abstract class only has methods, it could be an interface
> instead of an abstract class. Tracy, can you comment on why you made it an
> abstract class instead of an interface?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siostechcorp/exercises/pull/2/files#r24173904>.
>